### ISSUE TITLE
[Feature Request] Per-system input settings

### ISSUE THREAD

**Issue by Shideravan** [ares-emulator/ares#173]
Probably the most requested feature in ares.
It's better to proper document this.

**Comment by CRTRevivalDreamer** [comment]
I also want that feature to be implemented, but @Shideravan seems to be working on it.
Let's be patient...

**Comment by nanitaro** [comment]
I would like to report: When configuring gamepads from different consoles, the settings conflicts, not being separate one setting for each console.

For example, RetroArch allows you to set controls for each core and each game.

**Comment by vtmarik** [comment]
Yeah, especially when it comes to functionality within the ColecoVision/WonderSwan/etc. If I assign the numbers to my numpad, that shouldn't rewrite the controls for any other console.

Additionally, the Wonderswan button layouts (x#, y#) would be better served as being mappable to the d-pad and a stick vs. the shoulder buttons and d-pad since they are laid out in a similar manner to directionals.

I understand it was done with simplicity in mind, making everything try to map cleanly to an xbox-style controller, but simplicity should be done while maintaining practicality not in place of such.

**Comment by randomsomeone69** [comment]
I wonder if the configuration of the mapped controller are stored in the same file. It could be possible if the program can store this in multiple files.

**Comment by LukeUsher** [comment]
Copy/paste from discord:

> Honestly I think we do need per-system and per-game inputs, and in the UI we don't need to change much to allow it to happen.
> 
> We can keep the virtual pad system exactly as is, which you can map once and everything just works; users who like the current mapping system don't need to change anything, just map your virtual pads and be done with it, everything works as it does now.
> 
> When selecting a system in the dropdown, instead of it being a "view" of the virtual pad, it should be overrides;  when an override is not set, they should be a muted colour, showing the virtual mapping view so you know what it is mapped to, but that it's not set for this system specifically.
> 
> When configuring overrides, there should be an option to save for the current system, or just the current game.
> 
> When a game specific override is set, there should be an extra label warning you that a game specific config is in use, with an option to delete it.
> there might be improvements to be made but that's my initial thinking
and it can support the scenarios most people seem to want while still allowing a single global config

**Comment by rrrcabral** [comment]
I need that too.
Already on the second console that I decided to use Ares I noticed that it ruined the configuration of the previous console.
and now that I want to use Ares for a third console (as it is the most compatible, congratulations) I know that I will encounter a stressful situation of seeing that the new settings for this system will ruin the previous ones, and I suspect that I will have to abort the use of Ares and use another specific emulator for this third console.

[ just SEO words to help people find this issue: ares controllers multiple system ]

**Comment by kathyrollo** [comment]
@rrrcabral 
The dirty workaround right now is to have multiple copies of  ares. I previously had `Ares - Nintendo` and `Ares - Sega` folders.
I have since moved to Mesen for Nintendo, but still keep are for Sega. Mesen has a good controller mapping config per console.

**Comment by kathyrollo** [comment]
@rrrcabral
The dirty workaround right now is to have multiple copies of ares. I previously had Ares - Nintendo and Ares - Sega folders.

I have since moved to [Mesen](https://www.mesen.ca/) for Nintendo, but still keep Ares for Sega. Mesen has a sophisticated controller mapping scheme per console. Any good multi-system emulator should and already do. Don't know why the devs of Ares resent the idea so much.

**Comment by neotins** [comment]
I have NES, SNES, Genesis, N64 and Xbox style controllers all available to be used on my PC. The lack of separate controller support per system is a killer for me. As one of the top emulators available, I'm surprised that something like this was overlooked. Hopefully they'll add this in at some point, but until then, I'll have to use something else.

**Comment by LukeUsher** [comment]
added in 14f4fc0
